### PR TITLE
 soc: cc13x2_cc26x2: support setting CCFG values via Kconfig

### PR DESCRIPTION
**Pull request by bwitherspoon**
Several customer configuration (CCFG) settings are useful for development and deployment. This commit adds support for setting CCFG values via Kconfig. It follows TI's recommendation of setting CCFG values and then including the TI provided CCFG. This commit includes support for enabling the ROM bootloader and additional settings can be added as needed.

The ROM bootloader on the CC13x2 / CC26x2 LaunchPad development boards has been enabled by default. The bootloader can be started by holding BTN-1 during reset. This is the default value on POR, but the TI provided CCFG disables it. This can be useful during development to recover a board if DAPs get disabled.

### CODE REVIEW

**Comment by bwitherspoon** [comment]
The following is some verification that the resulting binary is as expected.

The CCFG section for the cc1352r1_launchxl board with the bootloader disabled:

```
$ZEPHYR_SDK_INSTALL_DIR/arm-zephyr-eabi/bin/arm-zephyr-eabi-objdump -s -j .ti_ccfg build/zephyr/zephyr.elf

build/zephyr/zephyr.elf:     file format elf32-littlearm

Contents of section .ti_ccfg:
 57fa8 00008001 100080ff fdff5800 3affbbf3  ..........X.:...
 57fb8 ffffffff ffffffff ffffffff ffffffff  ................
 57fc8 ffffffff ffffffff ffffffff ffffffff  ................
 57fd8 00ffff00 ffffffff 00ffffff 00c5c5ff  ................
 57fe8 000000ff 00000000 ffffffff ffffffff  ................
 57ff8 ffffffff ffffffff                    ........
```

The CCFG section for the cc1352r1_launchxl board with the bootloader enabled:

```
$ZEPHYR_SDK_INSTALL_DIR/arm-zephyr-eabi/bin/arm-zephyr-eabi-objdump -s -j .ti_ccfg build/zephyr/zephyr.elf

build/zephyr/zephyr.elf:     file format elf32-littlearm

Contents of section .ti_ccfg:
 57fa8 00008001 100080ff fdff5800 3affbbf3  ..........X.:...
 57fb8 ffffffff ffffffff ffffffff ffffffff  ................
 57fc8 ffffffff ffffffff ffffffff ffffffff  ................
 57fd8 c50ffec5 ffffffff 00ffffff 00c5c5ff  ................
 57fe8 000000ff 00000000 ffffffff ffffffff  ................
 57ff8 ffffffff ffffffff
```

The BL_CONFIG at 57fd8 looks correct when comparing to TRM section 11.2.1.13 and accounting for byte order of objdump output.

The following python script initializes the bootloader and can be used to test that the bootloader is indeed enabled on the LaunchPads after holding BTN-1 during reset (depends on [pyserial](https://pythonhosted.org/pyserial/)): [bootloader.py](https://github.com/zephyrproject-rtos/zephyr/files/3178520/bootloader.py.txt)

**Comment by galak** [comment]
Looks good to me, but will let @vanti review/comment.

**Review by vanti — CHANGES_REQUESTED** [review]
Looks good. Just a couple of doc comments.

File: boards/arm/cc1352r1_launchxl/doc/index.rst [hunk 1]
```diff
@@ -161,6 +161,14 @@ You can debug an application in the usual way.  Here is an example for the
    :maybe-skip-config:
    :goals: debug
 
+Bootloader
+==========
+
+The ROM serial bootloader on the CC13xx / CC26xx devices is enabled by default.
+The bootloader will start if there is no valid application image in flash or the
+so-called backdoor is enabled and BTN-1 is held down during reset. See the `TI
+CC2538 / CC26xx Serial Bootloader Interface`_ documentation for additional
```
[the remark this answers is not in the document]
Comment (bwitherspoon) [reply boards/arm/cc1352r1_launchxl/doc/index.rst]: You're right. I will make these changes.

**Review by vanti — APPROVED** [review]

File: boards/arm/cc1352r1_launchxl/doc/index.rst [hunk 2]
```diff
@@ -161,6 +161,14 @@ You can debug an application in the usual way.  Here is an example for the
    :maybe-skip-config:
    :goals: debug
 
+Bootloader
+==========
+
+The ROM bootloader on the CC13x2 / CC26x2 devices is enabled by default. The
```
Comment (dbkinder) [inline boards/arm/cc1352r1_launchxl/doc/index.rst]:
```suggestion
The ROM bootloader on CC13x2 and CC26x2 devices is enabled by default. The
```

File: boards/arm/cc1352r1_launchxl/doc/index.rst [hunk 3]
```diff
@@ -161,6 +161,14 @@ You can debug an application in the usual way.  Here is an example for the
...
+bootloader will start if there is no valid application image in flash or the
+so-called backdoor is enabled and BTN-1 is held down during reset. See the
+bootloader documentation in section ten of the `TI CC13x2 / CC26x2 Technical
```
Comment (dbkinder) [inline boards/arm/cc1352r1_launchxl/doc/index.rst]:
```suggestion
bootloader documentation in chapter 10 of the `TI CC13x2 / CC26x2 Technical
```

File: boards/arm/cc1352r1_launchxl/doc/index.rst [hunk 4]
```diff
@@ -161,6 +161,14 @@ You can debug an application in the usual way.  Here is an example for the
    :maybe-skip-config:
    :goals: debug
 
+Bootloader
+==========
+
+The ROM bootloader on the CC13x2 / CC26x2 devices is enabled by default. The
+bootloader will start if there is no valid application image in flash or the
+so-called backdoor is enabled and BTN-1 is held down during reset. See the
```
Comment (dbkinder) [inline boards/arm/cc1352r1_launchxl/doc/index.rst]:
```suggestion
so-called backdoor is enabled (via option
:option:`CONFIG_CC13X2_CC26X2_BOOTLOADER_BACKDOOR_ENABLE`)
and BTN-1 is held down during reset. See the
```

**Review by dbkinder — CHANGES_REQUESTED** [review]
add references to the config options and tweak the reference to the TRM

**Comment by bwitherspoon** [comment]
Rebased on master.

**Review by dbkinder — APPROVED** [review]
doc changes LGTM, thanks.

**Pull request merged**